torch/nn/modules/rnn.py

```diff
@@ -369,6 +370,16 @@ def __init__(self, *args, **kwargs):
         super(RNN, self).__init__(mode, *args, **kwargs)
 
 
+# XXX: LSTM and GRU implementation is different from RNNBase, this is because:
+# 1. we want to support nn.LSTM and nn.GRU in TorchScript and TorchScript in
+#    its current state could not support the python Union Type or Any Type.
+# 2. TorchScript static typing does not allow a Function or Callable type in
+#    Dict values, so we have to separately call _VF instead of using _rnn_impls
+# 3. This is temporary only and in the transition state that we want to make it
+#    on time for the release
+#
+# TODO: remove the overriding implementations for LSTM and GRU when TorchScript
```

Comment:
Could you remove the mention of Unions in this section? That is not the only way we could make this more expressible, and we haven't as a team made any promises about supporting it. Same thing with Any Type.